task_sdk/src/airflow/sdk/api/client.py

```diff
@@ -157,6 +158,12 @@ def get(self, key: str) -> VariableResponse:
         resp = self.client.get(f"variables/{key}")
         return VariableResponse.model_validate_json(resp.read())
 
+    def set(self, msg: PutVariable):
+        """Set an Airflow Variable via the API server."""
+        key = msg.key
+        self.client.put(f"variables/{key}", content=msg.model_dump_json())
```

Comment:
This should fail! `PutVariable` also has `type: Literal["PutVariable"] = "PutVariable"` which won't be recognizable by the APIServer.

What you want to send is `VariablePostBody`

Reply:
You can do something similar to:

https://github.com/apache/airflow/blob/cc503aad6bd99e0008a19fe2b2eaedec4411a871/task_sdk/src/airflow/sdk/api/client.py#L132

```py
key = msg.key
body = VariablePostBody(**msg.model_dump(exclude_unset=True))
self.client.put(f"variables/{key}", content=body)
```

I don't love recreating the class but if you have any other ideas feel free to explore

Reply:
I tested it before implementing this part in this way. As per pydantic, the extra fields by default are just blatantly ignored. Check here: https://docs.pydantic.dev/latest/concepts/models/#extra-fields

Example:
![image](https://github.com/user-attachments/assets/c8af7cc5-6111-4e34-a054-74e732796dfc)

Reply:
The API should forbid extra fields and that can be done by adding this `model_config = ConfigDict(extra='forbid')` in the model definition. I added it and tested it.

Works as expected:
![image](https://github.com/user-attachments/assets/04fcb732-e9a5-428c-b964-22410dbc3685)

I will push this change as a fix in this PR itself

Reply:
```
key = msg.key
body = VariablePostBody(**msg.model_dump(exclude_unset=True))
self.client.put(f"variables/{key}", content=body)
```

This seems cleaner than most of the other solutions I can think of, or we can just pass them as individual parameters which is messier, instead its better to pass them in an encapsulated manner.

Reply:
I have handled this part better now